openmetadata-service/src/main/java/org/openmetadata/service/search/SearchRepository.java

```diff
@@ -0,0 +1,406 @@
+          updated.getOwner());
+    }
+  }
+
+  default void handleDomainUpdates(EntityInterface original, EntityInterface updated, String eventType) {
+    if (eventType.equalsIgnoreCase("added")) {
+      this.updateSearchChildrenUpdated(
+          updated,
+          getDomainChangeScript(eventType, ""),
+          updated.getEntityReference().getType() + ".id",
+          ENTITY_TO_CHILDREN_MAPPING.get(updated.getEntityReference().getType()).toString(),
+          updated.getId().toString(),
+          updated.getDomain());
+    } else if (eventType.equalsIgnoreCase("updated")) {
+      this.updateSearchChildrenUpdated(
+          updated,
+          getDomainChangeScript(eventType, original.getDomain().getId().toString()),
+          updated.getEntityReference().getType() + ".id",
+          updated.getId().toString(),
+          ENTITY_TO_CHILDREN_MAPPING.get(updated.getEntityReference().getType()).toString(),
+          updated.getDomain());
+    }
+    if (eventType.equalsIgnoreCase("deleted")) {
+      this.updateSearchChildrenUpdated(
+          updated,
+          getDomainChangeScript(eventType, original.getDomain().getId().toString()),
+          updated.getEntityReference().getType() + ".id",
+          updated.getId().toString(),
+          ENTITY_TO_CHILDREN_MAPPING.get(updated.getEntityReference().getType()).toString(),
+          updated.getDomain());
+    }
+  }
+
+  default void handleClassificationUpdate(Classification entity) {
+    this.updateSearchEntityUpdated(entity, String.format(CLASSIFICATION_DISABLE_SCRIPT, entity.getDisabled()), "");
+  }
+
+  default String getOwnerChangeScript(String eventType, String ownerId) {
+    String scriptTxt = "";
+    if (eventType.equals("added")) {
+      scriptTxt = "if(ctx._source.owner == null){ ctx._source.put('owner', params)}";
+    } else if (eventType.equals("deleted")) {
+      scriptTxt = String.format("if(ctx._source.owner.id == '%s'){ ctx._source.remove('owner')}", ownerId);
+    } else if (eventType.equals("updated")) {
+      scriptTxt = String.format("if(ctx._source.owner.id == '%s'){ ctx._source.put('owner', params)}", ownerId);
+    }
+    return scriptTxt;
+  }
+
+  default String getDomainChangeScript(String eventType, String domainId) {
+    String scriptTxt = "";
+    if (eventType.equals("added")) {
+      scriptTxt = "if(ctx._source.domain == null){ ctx._source.put('domain', params)}";
+    } else if (eventType.equals("deleted")) {
+      scriptTxt = String.format("if(ctx._source.domain.id == '%s'){ ctx._source.remove('domain')}", domainId);
+    } else if (eventType.equals("updated")) {
+      scriptTxt = String.format("if(ctx._source.domain.id == '%s'){ ctx._source.put('domain', params)}", domainId);
+    }
+    return scriptTxt;
+  }
+
+  default void handleEntityDeleted(EntityInterface entity) {
+    switch (entity.getEntityReference().getType()) {
+      case Entity.DOMAIN:
```

Comment:
why we need this switch case, can't we make this generic code? we don't want to keep adding to this whenever we add a new entity right?